system-addon/lib/TopStoriesFeed.jsm

```diff
@@ -293,7 +291,7 @@ this.TopStoriesFeed = class TopStoriesFeed {
       return;
     }
 
-    if (this.newTabsSinceSpoc === 0 || this.newTabsSinceSpoc === this.spocsPerNewTabs) {
+    if (Math.random() <= this.spocsPerNewTabs) {
```

Comment:
nit: please add a comment to briefly describe `this.spocsPerNewTabs` in the `updateSettings()`

Reply:
Comment added. Thanks!